File: src/main/java/com/chat/app/service/UserService.java
package com.chat.app.service;

import com.chat.app.model.Users;

/**
 * The Interface UserService.
 */
public interface UserService {
	
	/**
	 * Find user by username.
	 *
	 * @param username the username
	 * @return the users
	 */
	public Users findUserByUsername(String username);
	
	/**
	 * Save user.
	 *
	 * @param user the user
	 * @return 
	 */
	public Users saveUser(Users user);
	
	/**
	 * Delete user.
	 *
	 * @param user the user
	 */
	public void deleteUser(Users user);
	
	/**
	 * Update user.
	 *
	 * @param username the username
	 * @param user the user
	 */
	public Users updateUser(String username, Users user);
	
	/**
	 * Authenticate user.
	 *
	 * @param username the username
	 * @param password the password
	 * @return boolean - true if success and false if failed
	 */
	public boolean authenticate(String username, String password);
}

File: src/main/resources/static/app-services/user.service.js
(function () {
    'use strict';

    angular
        .module('app')
        .factory('UserService', UserService);

    UserService.$inject = ['$http'];
    function UserService($http) {
        var service = {};

        //service.GetAll = GetAll;
        service.GetByUsername = GetByUsername;
        service.Create = Create;
        service.Update = Update;
        service.Delete = Delete;

        return service;

        //function GetAll() {
           // return $http.get('http://localhost:8080/api/users').then(handleSuccess, handleError('Error getting all users'));
        //}

        function GetByUsername(username) {
            return $http.get('http://localhost:8888/api/users/' + username).then(handleSuccessData, handleError('Error getting user by id'));
        }

      
        function Create(user) {
            return $http.post('http://localhost:8888/api/register', user).then(handleSuccess, handleError('Error creating user'));
        }

        function Update(user) {
            return $http.put('http://localhost:8888/api/update/' + user.username, user).then(handleSuccess, handleError('Error updating user'));
        }

        function Delete(username) {
            return $http.get('http://localhost:8888/api/delete/' + username).then(handleSuccess, handleError('Error deleting user'));
           /* .then(function OnSuccess(success) {
        		var response = { success: true };
               callback(response);
            },
            function onError(error){
            	var response = { success: false, message: 'Unable to delete the user' };
            	callback(response)
            });*/
        }

        // private functions
        
        function handleSuccessData(res) {
        	return res.data;          
        }
        
        function handleSuccess(res) {
        	return { success: true };          
        }

        function handleError(error) {
            return function () {
                return { success: false, message: error };
            };
        }
    }

})();
